reliability/single-fit/okhttp/src/test/kotlin/com/datadog/android/okhttp/HeadBasedSamplingTest.kt

```diff
@@ -0,0 +1,663 @@
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+    ExtendWith(TestConfigurationExtension::class)
+)
+@ForgeConfiguration(OkHttpConfigurator::class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class HeadBasedSamplingTest {
+
+    private lateinit var stubSdkCore: StubSDKCore
+    private lateinit var mockServer: MockWebServer
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        stubSdkCore = StubSDKCore(forge)
+        val registry: Any = Datadog::class.java.getStaticValue("registry")
+        val instances: MutableMap<String, SdkCore> = registry.getFieldValue("instances")
+        instances += stubSdkCore.name to stubSdkCore
+        mockServer = MockWebServer()
+
+        val fakeTraceConfiguration = TraceConfiguration.Builder().build()
+        Trace.enable(fakeTraceConfiguration, stubSdkCore)
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        unregisterGlobalTracer()
+        Datadog.stopInstance(stubSdkCore.name)
+        mockServer.shutdown()
+    }
+
+    @Test
+    fun `M use network sampling rate W call is made { no parent context }`(forge: Forge) {
+        // Given
+        if (forge.aBool()) {
+            // outcome should be the same for cases where there is GlobalTracer and when local tracer is used
+            registerGlobalTracer(0.0)
+        }
+
+        mockServer.enqueue(MockResponse())
+        mockServer.start()
+        val okHttpClient = OkHttpClient.Builder()
+            .addInterceptor(
+                TracingInterceptor.Builder(
+                    tracedHostsWithHeaderType = mapOf(mockServer.hostName to setOf(TracingHeaderType.DATADOG))
+                )
+                    .setTraceContextInjection(TraceContextInjection.All)
+                    .setSdkInstanceName(stubSdkCore.name)
+                    .setTraceSampleRate(100f)
+                    .build()
+            )
+            .build()
+
+        // When
+        okHttpClient.newCall(
+            Request.Builder()
+                .url(mockServer.url("/"))
+                .build()
+        ).execute()
+
+        // Then
+        val requestSent = mockServer.takeRequest()
+        assertThat(requestSent.getHeader(DATADOG_SAMPLING_PRIORITY_HEADER)).isEqualTo("1")
+        val leastSignificantTraceId = requestSent.getHeader(DATADOG_TRACE_ID_HEADER)
+        checkNotNull(leastSignificantTraceId)
+        val spanId = requestSent.getHeader(DATADOG_SPAN_ID_HEADER)
+        checkNotNull(spanId)
+        val datadogTags = requestSent.getHeader(DATADOG_TAGS_HEADER)
+        checkNotNull(datadogTags)
+        assertThat(datadogTags).startsWith(MOST_SIGNIFICANT_TRACE_ID_KEY)
```

Comment:
> [!WARNING]
> Using `startsWith` makes an assumption on the order of the tags in the header. Maybe we should check for the presence of the `MOST_SIGNIFICANT_TRACE_ID_KEY` key but not that it'll be first?

> [!NOTE]
> Are there other tags that we expect and should check here too?

Reply:
Good point, I'll update this verification logic.

> Are there other tags that we expect and should check here too?

No, by looking on the `DatadogHttpCodec` implementation only `_dd.p.tid` key/value is set for the `x-datadog-tags` header.